hvsock.go

```diff
@@ -65,18 +140,43 @@ func (addr *HvsockAddr) fromRaw(raw *rawHvsockAddr) {
 	addr.ServiceID = raw.ServiceID
 }
 
+// Sockaddr interface allows use with `sockets.Bind()` and `.ConnectEx()`
+func (r *rawHvsockAddr) Sockaddr() (ptr unsafe.Pointer, len int32, err error) {
+	return unsafe.Pointer(r), int32(unsafe.Sizeof(rawHvsockAddr{})), nil
```

Comment:
Can we do the validations in this function itself instead of have the caller do it after this? Referring to the steps in validateSockAddr:

```go
ptr, n := unsafe.Pointer(r), int32(unsafe.Sizeof(rawHvsockAddr{})
if ptr == nil {
	return nil, 0, fmt.Errorf("pointer is %p: %w", ptr, ErrInvalidPointer)
}
if n < 1 {
	return nil, 0, fmt.Errorf("buffer size %d < 1: %w", n, ErrBufferSize)
}
return ptr, n, nil

Reply:
I guess for the buffer size, unless we completely butchered a change and removed every field in rawHvsockAddr we'd never hit that

Reply:
my intention was that since `(*rawHvsockAddr) Sockaddr()` was user defined, and internally we could have a bug in how we define it, `validateSockAddr` is always going to run and prevent segfaults or panics
But, looking at it now, I guess we can use reflection to get the pointer and struct size, like how `json.Unmarshall` does it ....